contracts/core/Allo.sol

```diff
@@ -151,104 +170,55 @@ contract Allo is Transfer, Initializable, Ownable, AccessControl {
     /// ==== External/Public Functions =====
     /// ====================================
 
-    /// @notice Creates a new pool (with clone for approved strategies)
+    /// @notice Creates a new pool (with custom strategy)
     /// @param _identityId The identityId of the pool
-    /// @param _allocationStrategy The address of the allocation strategy
-    /// @param _initAllocationData The data to initialize the allocation strategy
-    /// @param _cloneAllocationStrategy Whether to clone the allocation strategy
-    /// @param _distributionStrategy The address of the distribution strategy
-    /// @param _initDistributionData The data to initialize the distribution strategy
-    /// @param _cloneDistributionStrategy Whether to clone the distribution strategy
+    /// @param _strategy The address of strategy
+    /// @param _initStrategyData The data to initialize the strategy
     /// @param _token The address of the token
     /// @param _amount The amount of the token
     /// @param _metadata The metadata of the pool
     /// @param _managers The managers of the pool
-    function createPoolWithClone(
+    function createPoolWithCustomStrategy(
         bytes32 _identityId,
-        address _allocationStrategy,
-        bytes memory _initAllocationData,
-        bool _cloneAllocationStrategy,
-        address _distributionStrategy,
-        bytes memory _initDistributionData,
-        bool _cloneDistributionStrategy,
+        address _strategy,
+        bytes memory _initStrategyData,
         address _token,
         uint256 _amount,
         Metadata memory _metadata,
         address[] memory _managers
     ) external payable returns (uint256 poolId) {
-        address allocationStrategy;
-        address distributionStrategy;
-
-        // DISCUSS: We either
-        // - allow cloning of every contract (use the bool)
-        // - or we force cloning of only approved contracts
-
-        if (_allocationStrategy == address(0) || _distributionStrategy == address(0)) {
+        if (_strategy == address(0)) {
```

Comment:
need a check here to ensure the _strategy is not an approved one. We don't want to use those

Reply:
We should be set for all approved ones to make sure the `initialize()` call fails, but added this check for extra safety.